### PR TITLE
Search: update search hook placement

### PR DESCRIPTION
**Pull request by alexisgrow**
Move custom hook from `onQuerySubmit` to `search`. This means that the hook is invoked whenever a search is conducted. Since the hook has not been in any releases, we're free to change its name without any backwards compatibility issues. Changed from `onSubmit` to `onConductSearch`.

TEST=manual,unit

Test with Overlay locally. Confirm that this hook is invoked when searches are conducted via typing in the search bar as well as with the ANSWERS.search function.

### CODE REVIEW

File: src/ui/components/search/searchcomponent.js [hunk 1]
```diff
@@ -602,6 +600,7 @@ export default class SearchComponent extends Component {
     this._throttled = true;
     setTimeout(() => { this._throttled = false; }, this._searchCooldown);
 
+    this.customHooks.onConductSearch(query);
```
Comment (creotutar) [inline src/ui/components/search/searchcomponent.js]: Why here and not in the `search` function?

Comment (alexisgrow) [reply src/ui/components/search/searchcomponent.js]: Ah, good point 👀

**Review by nanhu95 — APPROVED** [review]
LGTM!

File: README.md [hunk 2]
```diff
@@ -461,9 +461,8 @@ ANSWERS.addComponent('SearchBar', {
   customHooks: {
     // Optional, a callback invoked when the clear search button is clicked
     onClearSearch: function() {},
-    // Optional, a function invoked when a search is submitted via this component. The users search
-    // is passed in as a string
-    onSubmit: function(searchTerms) {}
+    // Optional, a function invoked when a search is conducted. The search terms are passed in as a string
+    onConductSearch: function(searchTerms) {}
```
Comment (creotutar) [inline README.md:467]: This is probably why the wording was changed, but it's OK that this is called when we do a defaultInitialSearch in `ANSWERS.init` right? It's wanted behavior that this custom hook in the SearchBar component is called when that happens?

Comment (alexisgrow) [reply README.md:467]: Yes exactly! It should be called whenever a search is conducted - default initial search, ANSWERS.search or a search from typing in the search input!

**Review by creotutar — APPROVED** [review]
✨ ✨ ✨ LGTM

**Review by tmeyer2115 — APPROVED** [review]

**Pull request merged**